CRM/Utils/Token.php

```diff
@@ -1606,6 +1605,44 @@ protected static function _buildMembershipTokens() {
   }
 
   /**
+   * store event tokens on the static _tokens array
+   */
+  protected static function _buildEventTokens() {
+    $key = 'event';
+    if (!isset(self::$_tokens[$key]) || self::$_tokens[$key] == NULL) {
+      $eventTokens = array();
+      $tokens = CRM_Core_SelectValues::eventTokens();
+      foreach ($tokens as $token => $dontCare) {
+        $eventTokens[] = substr($token, (strpos($token, '.') + 1), -1);
+      }
+      $customtokens = CRM_CORE_BAO_CustomField::getFields('Event');
+      foreach ($customtokens as $tokenkey => $tokenvalue) {
+        $eventTokens[] = 'custom_' . $tokenkey;
+      }
+      self::$_tokens[$key] = $eventTokens;
+    }
+  }
+
+  /**
+   * store  participant tokens on the static _tokens array
+   */
+  protected static function _buildParticipantTokens() {
+    $key = 'participant';
+    if (!isset(self::$_tokens[$key]) || self::$_tokens[$key] == NULL) {
+      $participantTokens = array();
+      $tokens = CRM_Core_SelectValues::participantTokens();
+      foreach ($tokens as $token => $dontCare) {
+        $participantTokens[] = substr($token, (strpos($token, '.') + 1), -1);
+      }
+      $customtokens = CRM_CORE_BAO_CustomField::getFields('Participant');
```

Comment:
Pretty sure it needs to be `CRM_Core_BAO_CustomField` (i.e. `Core` in mixed case).